video/out/vo.c

```diff
@@ -944,6 +944,7 @@ static bool render_frame(struct vo *vo)
     } else {
         in->rendering = true;
         in->hasframe_rendered = true;
+        in->request_redraw = false;
```

Comment:
This clears `in->request_redraw` unconditionally, but originally it was only cleared if the frame was not dropped. I think it's a (new) bug.

Reply:
Not true, see line 941.

Reply:
Yeah, you're totally right, I didn't spot that. Thanks a lot. I force-pushed a change that I believe addresses this. The logic feels very subtle so I tried to explain what's going on by adding a comment to the code.